pkg/auth/auth.go

```diff
@@ -19,6 +19,9 @@ import (
 	terminal "golang.org/x/term"
 )
 
+// ErrUnauthorizedForCredentials is returned when the status code returned is 401.
+var ErrUnauthorizedForCredentials = errors.New("invalid username/password")
```

Comment:
Should this be more specific, somehow? “user-entered credentials are invalid”, i.e. we don’t use this to report that the pre-existing credentials in `auth.json` are invalid if user provided a working set…

I’m not too worried, c/common can change the API at any time. It’s mostly that the “when the status come returned is 401” documentaiton seems way too focused on the implementation details vs. the caller-relevant semantics.

Reply:
I share the desire to massage the error message but at the moment only focus on https://github.com/containers/podman/pull/17581. 

Changing the error message will likely break tests in Skopeo, Buildah and Podman.

Reply:
FWIW I care very little about the _text_; I was focusing on the as _meaning_ of the new typed error introduced by this PR. But, again, the API is not stable so this doesn’t need to be perfect.

Reply:
Aaaah, thanks!  Feel free to shoot an alternative name/error message.

Reply:
```go
// ErrNewCredentialsInvalid means that the new user-provided credentials are not accepted by the registry
```
? Something vaguely in that direction.